test/e2e/proxy_test.go

```diff
@@ -15,25 +16,63 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/openshift/windows-machine-config-operator/controllers"
+	"github.com/openshift/windows-machine-config-operator/pkg/cluster"
 	"github.com/openshift/windows-machine-config-operator/pkg/patch"
 	"github.com/openshift/windows-machine-config-operator/pkg/retry"
+	"github.com/openshift/windows-machine-config-operator/pkg/windows"
 )
 
 // proxyTestSuite contains the validation cases for cluster-wide proxy.
 // All subtests are skipped if a proxy is not enabled in the test environment.
 func proxyTestSuite(t *testing.T) {
+	if !cluster.IsProxyEnabled() {
+		t.Skip("cluster-wide proxy is not enabled in this environment")
+	}
+
+	t.Run("Trusted CA ConfigMap validation", testTrustedCAConfigMap)
+	t.Run("Environment variables validation", testEnvVars)
+}
+
+// testEnvVars tests that on each node
+// 1. the system-level environment variables are set properly as per the cluster-wide proxy
+// 2. the required Windows services pick up the proper values for proxy environment variables
+func testEnvVars(t *testing.T) {
 	tc, err := NewTestContext()
 	require.NoError(t, err)
 
 	clusterProxy, err := tc.client.Config.ConfigV1().Proxies().Get(context.TODO(), "cluster", meta.GetOptions{})
 	if err != nil {
 		require.NoError(t, err)
 	}
-	if clusterProxy.Status.HTTPProxy == "" && clusterProxy.Status.HTTPSProxy == "" {
-		t.Skip("cluster-wide proxy is not enabled in this environment")
-	}
+	expectedEnvVars := map[string]string{}
+	expectedEnvVars["HTTP_PROXY"] = clusterProxy.Status.HTTPProxy
+	expectedEnvVars["HTTPS_PROXY"] = clusterProxy.Status.HTTPSProxy
+	expectedEnvVars["NO_PROXY"] = clusterProxy.Status.NoProxy
+	err = tc.waitForWindowsNodes(gc.numberOfMachineNodes, false, false, false)
+	require.NoError(t, err, "timed out waiting for Windows Machine nodes")
+	require.Greater(t, len(gc.allNodes()), 0, "test requires at least one Windows node to run")
+	for _, node := range gc.allNodes() {
+		t.Run(node.GetName(), func(t *testing.T) {
+			addr, err := controllers.GetAddress(node.Status.Addresses)
+			require.NoError(t, err, "unable to get node address")
 
-	t.Run("Trusted CA ConfigMap validation", testTrustedCAConfigMap)
+			for _, proxyVar := range cluster.AcceptableProxyVars {
+				value, err := tc.getSystemEnvVar(addr, proxyVar)
+				require.NoError(t, err, "unable to get value of %s from instance", proxyVar)
+				assert.Equalf(t, expectedEnvVars[proxyVar], value, "incorrect value for %s", proxyVar)
+			}
+
+			for _, svcName := range windows.RequiredServices {
```

Comment:
Are we adding the proxy variables to the configuration all services? Where are we doing that?

Reply:
We set the ENV vars at machine level and they get propogated to the services. We are just verifying they are set correctly in the e2e test as having them at machine and service level is important.
PTAL https://github.com/openshift/enhancements/blob/4f18064f4ad9124efec465c59346e37171c0fcc9/enhancements/windows-containers/windows-node-egress-proxy.md?plain=1#L177

Reply:
I completely forgot about this. Sorry about that. I suggest adding this as a comment somewhere in `reconcileEnvironment()`